logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/GetOrRegisterClientUseCase.kt

```diff
@@ -53,8 +58,7 @@ class GetOrRegisterClientUseCaseImpl(
                         is VerifyExistingClientResult.Success -> RegisterClientResult.Success(result.client)
                         is VerifyExistingClientResult.Failure.Generic -> RegisterClientResult.Failure.Generic(result.genericFailure)
                         is VerifyExistingClientResult.Failure.ClientNotRegistered -> {
-                            clearClientData()
-                            clientRepository.clearRetainedClientId()
+                            clearOldClientRelatedData()
```

Comment:
Are we guaranteed that the currentClientId hasn't already been cached in the user session at this point?

Reply:
It shouldn't happen because this use case should be used only when there's no registered client id yet, but I added clearing it just in case. @typfel do you think that's enough, we could collect `observeCurrentClientId` in `init` and update `_clientId` dynamically 🤔

Reply:
👍 should be good then!